android/src/main/java/com/rivereactnative/RiveReactNativeView.kt

```diff
@@ -108,6 +114,44 @@ class RiveReactNativeView(private val context: ThemedReactContext) : FrameLayout
     addView(riveAnimationView)
   }
 
+  @OptIn(ControllerStateManagement::class)
+  override fun onAttachedToWindow() {
+    // The below solves: https://github.com/rive-app/rive-react-native/issues/198
+    // When the view is returned to, we reuse the view's resources.
+    // For example, navigating to a new page and returning, or a TabView.
```

Comment:
Oh I guess this is due to `RiveReactNativeView` being still alive.
Yeah, with `RiveAnimationView` we currently make sure that all resources get cleared out and fully re-initialized `onAttach`. Restoring a controller is indeed more efficient because you don't have to rebuild a bunch of resources.

To handle these lifecycle situations you can also use a `LifecycleObserver`, which we use in `RiveAnimationView`

Reply:
@umberto-sonnino  Exactly this! I was initially thinking of using LifecycleObserver but I can't override the default Rive LifecycleObserver and add additional behaviour on to that (unless I missed something). And longer term I wouldn't want these two implementation to get out of sync, for example if Rive Android adds additional logic to it.

I was thinking that instead Android should move its `acquire` call to the LifecycleObserver's onCreate, as it's calling `release` in dispose.